### ISSUE TITLE
rtsph264 server hang/stop

### ISSUE THREAD

**Issue by luisbarreto** [EliasKotlyar/Xiaomi-Dafang-Hacks#304]
Hi,

I'm using custom firmware with a Dafang camera integrated with Home Assistant.
In HA I get the thumbnail no more than some minutes (don't know exactly because it just seem to hang randomly.
The thumbnail from motion detection works in HA home screen.
In HA (just after starting the camera - i.e. camera not hung) if I click the video thumbnail, I can't get the live video feed. Even waiting for more than a couple of minutes...

I am using mqtt autodiscovery in HA (also configured in camera), but for the video stream I am using the following in configuration.yaml

```
camera:
  - platform: ffmpeg
  name: dafang
  input: -rtsp_transport tcp -i rtsp://CAMERA_IP:8554/unicast
```

I also downsized the video feed to 1280 x 720 in hope of less computing power needed and less bandwidth without any apparent improvement.
In short, video stops updating HA thumbnail after some time and live video feed (apparently) don't work. Usually if I start rtsp-h264 service, things start working again for a little long.

Can anyone advise?

**Comment by goermezer** [comment]
Same for me since 3 days.

**Comment by jmtatsch** [comment]
I guess our rtspserver settings of resolution, keyframing, Bitrate could use more tuning. If you are an expert, please contribute. Also Video handling with HomeAssitant in general could be more fluid.

**Comment by goermezer** [comment]
After a lot of trial and error, I found out, that on my cams the parameter -S in RTSPH264OPTS causes the issues. After changing it in /system/sdcard/config/rtspserver.conf to:

`RTSPH264OPTS="-W1280 -H736"`

it works stable now.

**Comment by jmtatsch** [comment]
Great catch. 
-S enables HLS & MPEG-DASH with segment duration in seconds (default 5)
We should remove that from the default rtspserver.conf as it does't work well anyways.

**Comment by jmtatsch** [comment]
Wasn't really a default but removed it from the example: https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/commit/a0fccc5cb7cecad65125cdd5ae0537943a2400a5
Don't know how it ended up in your and my config...

**Comment by luisbarreto** [comment]
Just let me add my most recent experience:

Maybe consequence of my tests I didn't had the -S parameter in `/system/sdcard/config/rtspserver.conf`, even so I still can't get the thumbnail for live camera in HA for even half a day. 

I did try and apparently I'm heaving success with rtsp-mjpeg.
Disabling rtsp-h264 and enabling this rtsp-mjpeg stopped the issues I was having. This way I have it running for four days in a row without issues. Also motion snapshot is apparently working as expected.
